### ISSUE TITLE
#line 0 with debug info: cannot have column info without line info

### ISSUE THREAD

**Issue by DimitryAndric** [llvm/llvm-project#56186]
As reported in https://bugs.freebsd.org/264853, compiling peg/leg (see https://www.piumarta.com/software/peg/) with clang and debug info turned on results in fatal backend errors:

```
cannot have column info without line info
!1417 = distinct !DILexicalBlock(scope: !1418, file: !9, column: 8)
cannot have column info without line info
!1417 = distinct !DILexicalBlock(scope: !1418, file: !9, column: 8)
fatal error: error in backend: Broken module found, compilation aborted!
```

This seems to be caused by the `#line 0` directives in https://www.piumarta.com/software/peg/peg-0.1.18/src/peg.peg-c. As far as I can determine, a `#line` directive should have a "non-negative decimal integer constant" as parameter, so zero seems to be explicitly allowed? But it is not handled correctly in the debug info case.

Minimized test case:

```c
// clang -cc1 -triple x86_64-- -S -debug-info-kind=standalone peg-min.c
int a();

void b() {
# 0
  if (a())
    ;
}
```

**Comment by dwblaikie** [comment]
Fair enough. Looks like we should have an assertion in DILexicalBlock's ctor to catch this earlier and then somewhere in the call stack (I mean, DILexicalBlock could do it itself) the column should be zero'd or otherwise ignored.

**Comment by theo-lw** [comment]
Seems related to https://github.com/llvm/llvm-project/issues/51825.
